### PR TITLE
Jailhouse

### PR DESCRIPTION
**Pull request by glima**
@SebastianBoe any clues on why, for that board, running the target (make) run right away is not triggering the "all" one? That works for, say, qemu_x86.

### CODE REVIEW

**Comment by SebastianBoe** [comment]
@glima : I have no idea why this is working for qemu_x86.

PR to fix this: https://github.com/zephyrproject-rtos/zephyr/pull/5314

**Review by SebastianBoe — CHANGES_REQUESTED** [review]
After these patches are applied I am OK with this:

https://github.com/glima/zephyr/pull/1

**Comment by glima** [comment]
Thanks a lot, @SebastianBoe ! Let's see if the commit separation as you posted is fine for others, just then I'd change.

**Comment by glima** [comment]
@SebastianBoe I had to change some strings to those list-strings, otherwise \s would appear in the final expansion and qemu would cry (at least for my cmake version—3.10.0). make run right away for a Jailhouse build (cmake -DBOARD=x86_jailhouse -DJAILHOUSE_QEMU_IMG_FILE=something) on the synchronization example still does not compile things before firing up qemu.

File: cmake/qemu/CMakeLists.txt [hunk 1]
```diff
@@ -158,8 +172,7 @@ if(EMU_PLATFORM)
       ${QEMU_FLAGS}
       ${QEMU_EXTRA_FLAGS}
       ${MORE_FLAGS_FOR_${target}}
-      -kernel $<TARGET_FILE:${logical_target_for_zephyr_elf}>
-      DEPENDS ${logical_target_for_zephyr_elf}
```
Comment (SebastianBoe) [inline cmake/qemu/CMakeLists.txt]: We should not be removing this DEPENDS line.

**Review by SebastianBoe — CHANGES_REQUESTED** [review]

File: cmake/qemu/CMakeLists.txt (hunk 1 above)
Comment (SebastianBoe) [reply cmake/qemu/CMakeLists.txt]: I think keeping DEPENDS will fix this issue:

>make run right away for a Jailhouse build (cmake -DBOARD=x86_jailhouse -DJAILHOUSE_QEMU_IMG_FILE=something) on the synchronization example still does not compile things before firing up qemu.

Comment (glima) [reply cmake/qemu/CMakeLists.txt]: Sorry, merge oopsie. Will fix it.

**Review by tyrantlizard77 — APPROVED** [review]

**Review by SebastianBoe — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Pull request merged**